lib/src/simulations/Hypocycloid.dart

```diff
@@ -0,0 +1,455 @@
+            -2 *
+            pi) {
+      time = 0;
+      ys.clear();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.animating) {
+      WidgetsBinding.instance.addPostFrameCallback((_) => update());
+    }
+    if (!(widget.isLandscape ^ orientationChanged)) {
+      clearscreen();
+      orientationChanged = !orientationChanged;
+    }
+
+    return Container(
+      child: Transform.scale(
+        scale: widget.isLandscape
+            ? 0.7 * widget._scaleAmount
+            : widget._scaleAmount,
+        child: CustomPaint(
+          painter: NormalHypoCycloidPainter(
+            widget.outerRadius,
+            widget.innerRadius,
+            time,
+            (widget.isLandscape
+                    ? MediaQuery.of(context).size.width / 3
+                    : MediaQuery.of(context).size.width / 2)
+                .roundToDouble(),
+            (MediaQuery.of(context).size.height / 3).roundToDouble(),
+            widget.animate,
+            points,
+            context,
+          ),
+          child: Container(),
+        ),
+      ),
+    );
+  }
+}
+
+class NormalHypoCycloidPainter extends CustomPainter {
+  int innerRadius, outerRadius;
+  Offset smallCenter;
+  double transformx, transformy, time;
+  BuildContext context;
+  List<Offset> points = [];
+  bool animate;
+
+  NormalHypoCycloidPainter(
+    this.outerRadius,
+    this.innerRadius,
+    this.time,
+    this.transformx,
+    this.transformy,
+    this.animate,
+    points,
+    this.context,
+  );
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    Offset coor = new Offset(transformx, transformy);
+    if (!animate) {
+      var paint = Paint();
+      paint.style = PaintingStyle.stroke;
+      paint.strokeWidth = 2;
+      this.points.clear();
+      paint.color = Theme.of(context).accentColor;
+      canvas.drawCircle(
+          Offset(transformx, transformy), innerRadius.toDouble(), paint);
+      paint.color = Colors.red;
+      if (outerRadius != 0) {
+        for (double loopi = 0;
+            loopi <= (innerRadius / innerRadius.gcd(outerRadius)) * 2 * pi;
+            loopi += 0.01) {
+          this.points.add(Offset(
+                  (((innerRadius - outerRadius) * cos(loopi)) +
+                      (outerRadius *
+                          cos(((innerRadius / outerRadius) - 1) * loopi))),
+                  (((innerRadius - outerRadius) * sin(loopi)) -
+                      (outerRadius *
+                          sin(((innerRadius / outerRadius) - 1) * loopi))))
+              .translate(transformx, transformy));
+        }
+        canvas.drawPoints(PointMode.polygon, points, paint);
+      }
+    } else {
+      Paint paint = new Paint();
+      paint.style = PaintingStyle.stroke;
+      paint.strokeWidth = 2;
+      paint.color = Theme.of(context).accentColor;
+      canvas.drawCircle(coor, innerRadius.toDouble(), paint);
+      smallCenter = Offset((innerRadius + outerRadius) * cos(time),
```

Comment:
make it 
innerRadius - outerRadius
to display the rotating circle inside the base circle